DynamicBuffer/Ex1.hpp

```diff
@@ -64,60 +45,20 @@ DynamicBuffer<T>& DynamicBuffer<T>::operator=(DynamicBuffer<T>&& other) noexcept
 
     if (this != &other) {
 
-        m_buffer = other.m_buffer;
-        m_size = other.m_size;
-
-        other.m_buffer = nullptr;
-        other.m_size = 0;
+        m_buffer = std::move(other.m_buffer);
     }
 
     return *this;
 }
 
-template <typename T> //Dtor
-DynamicBuffer<T>::~DynamicBuffer() noexcept {
-
-    deAllocate();
-}
-
-
 template <typename T>
 T& DynamicBuffer<T>::operator[](int idx) {
 
-    if (idx < 0) {
```

Comment:
why did you remove the boundary check?
what would happen if idx is less than 0?
what would happen if idx is greater than `size`?

Reply:
totally right should be using the  "at" method, so there wont be undefined behavior